src/scene/mesh-instance.js

```diff
@@ -59,6 +59,85 @@ class Command {
     }
 }
 
+/**
+ * Internal helper class for storing the shader and related mesh bind group in the shader cache.
+ *
+ * @ignore
+ */
+class ShaderInstance {
```

Comment:
Should this go in its own module?

Reply:
I decided against it for now, as it's just internal to mesh instance, not even exported.